src/Kernel/Process.class.st

```diff
@@ -198,32 +198,17 @@ Process >> debug [
 	^ self debugWithTitle: 'Debug'.
 ]
 
-{ #category : #debugging }
-Process >> debug: context title: title [
-	"Open debugger on self with context shown on top"
-
-	^ self debug: context title: title full: false.
-
-]
-
-{ #category : #debugging }
-Process >> debug: context title: title full: bool [
-	
-	^ UIManager default
-		debugProcess: self 
-		context: context 
-		label: title 
-		fullView: bool
-]
-
 { #category : #debugging }
```

Comment:
Could this method packaged with Oups? because it continues to promote the dependency to UIManager?

Reply:
The problem is (at least) two-fold:
- Before **Oups**, everything went through the UIManager, but instead of a simple API the full logic was implemented there and the API was crappy. We need a centralized API somewhere, we need to devise where if not in the UIManager to put this API so that it is accessible to most of the code without introducing other dependencies.
- Now, the problem is that **Oups** depends on UIManager to know if it can open graphical debuggers. When we designed it, we knew we did not like it. However, we decided to keep that problem for the next steps. The goal would be to remove that dependency and be as independent as possible so that the debugger infrastructure could be loaded the soonest as possible in the system (ideally, in the bootstrap?). For now, I've identified that it could be loaded just after Kernel, if not for that UIManager dependency.
So the answer is yes, we need to remove that dependency. I advise we keep that problem for after :)